### ISSUE TITLE
ItemRepository leverages a var for prepared statements

### ISSUE THREAD

**Issue by WadeWaldron** [lagom/online-auction-scala#90]
The ItemRepository (and possibly others) leverages a var for prepared statements. This is contrary to the documentation which proposes that a promise be used instead:

https://www.lagomframework.com/documentation/1.4.x/scala/ReadSideCassandra.html#Prepare

An alternative approach would be to use a lazy val.

**Comment by fgfernandez0321** [comment]
Hi,
Using `lazy val` approach would be something like:
```
private lazy val insertItemCreatorStatement = session.prepare("""
        INSERT INTO itemCreator(
          itemId,
          creatorId
        ) VALUES (?, ?)
      """)
```
```
  private def insertItemCreator(item: Item) = {
    insertItemCreatorStatement map (_.bind(item.id, item.creator))
  }
```
I would like to help with this.

**Comment by TimMoore** [comment]
Thanks, @fgfernandez0321!

I think it would be best to align the documentation with the example. Right now, the documentation @WadeWaldron linked to uses promises. So, either this example should also be updated to use promises, or they should both be updated to use `lazy val`. Using `lazy val` looks simpler, but I'm not sure if there's some disadvantage when you don't invoke this from the `prepare` hook (such as a race condition or other concurrency bug). @patriknw I think you updated this documentation to use promises... do you know?

**Comment by fgfernandez0321** [comment]
Hi @TimMoore 

I tested `lazy val` approach in a different project and it works great however the documentation suggests another approach using promises and the `online-auction-scala` example should be aligned with it.

Using `promises` approach would be something like:
```
private val insertItemCreatorPromise = Promise[PreparedStatement]
private def insertItemCreator: Future[PreparedStatement] = insertItemCreatorPromise.future
```
```
private def prepareStatements() = {

    val insertItemCreatorFuture = session.prepare("""
        INSERT INTO itemCreator(itemId, creatorId) VALUES (?, ?)
      """)
    insertItemCreatorPromise.completeWith(insertItemCreatorFuture)

    val insertItemSummaryByCreatorFuture = session.prepare("""
        INSERT INTO itemSummaryByCreator(
          creatorId,
          itemId,
          title,
          currencyId,
          reservePrice,
          status
        ) VALUES (?, ?, ?, ?, ?, ?)
      """)
    insertItemSummaryByCreatorPromise.completeWith(insertItemSummaryByCreatorFuture)

    val updateItemSummaryStatusFuture = session.prepare("""
        UPDATE itemSummaryByCreator SET status = ? WHERE creatorId = ? AND itemId = ?
      """)
    updateItemSummaryStatusPromise.completeWith(updateItemSummaryStatusFuture)

    for {
      _ <- insertItemCreatorFuture
      _ <- insertItemSummaryByCreatorFuture
      _ <- updateItemSummaryStatusFuture
    } yield Done
  }
```
I have both approaches implemented using `ItemRepository` as example. 

Please let me know if I can collaborate with this.

**Comment by renatocaval** [comment]
Resolved by #95
